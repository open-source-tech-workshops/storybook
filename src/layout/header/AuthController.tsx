import * as React from 'react';
import { Button } from '@mui/material';
import { UserInfoDropDown } from './UserDropdown';

export interface AuthProps{
    'userData'?: UserProps
}
export interface UserProps{
    'id':string, 'name':string
}
export const AuthController = (props: AuthProps ) => {
    if(props.userData){
        return(
            <UserInfoDropDown/>
        );
    }else{
        return(
            <Button data-testid='login-btn' variant='contained'>Login</Button>
        );
    }
}