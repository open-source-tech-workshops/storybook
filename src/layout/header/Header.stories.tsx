import React, { ReactElement } from 'react';
import { storiesOf, addDecorator, ComponentMeta } from '@storybook/react';
import { muiTheme } from 'storybook-addon-material-ui';
import Header from './Header';

const authSearchOptions = [
  { label: 'The Godfather', id: 1 },
  { label: 'Pulp Fiction', id: 2 },
];
const searchFieldChangeHanlder = (value: string) => {
  console.log('changed', value);
};

storiesOf('MaterialUI', module)
  .addDecorator(muiTheme())
  .add('MainHeader', () => (<Header searchFieldHandler={searchFieldChangeHanlder} />));
