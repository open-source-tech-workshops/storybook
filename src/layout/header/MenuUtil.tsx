import * as React from 'react';
import { MenuItem, Menu, Button, Grid } from '@mui/material';
import { MouseEventHandler } from 'react';
import { HeaderProps } from './Header';

export interface MenuItemProps {
    menuDetail: { 'id': number, 'title': string, 'open'?: boolean, 'handler': MouseEventHandler, 'menuItems'?: MenuItemProps[] }
}

interface MenuProps extends MenuItemProps {
    'handleClose': MouseEventHandler,
}

export function MenuItemComponentList(props: MenuProps): React.ReactElement | null {
    const open = Boolean(props.menuDetail.open);
    if (props.menuDetail.menuItems && props.menuDetail.menuItems.length > 0) {
        let menuItemComponent: Array<React.ReactElement> = [];
        menuItemComponent = props.menuDetail.menuItems.map((item) => {
            return <MenuItem data-testid={'menu-item-' + props.menuDetail.id + '-' + item.menuDetail.id} onClick={item.menuDetail.handler} key={item.menuDetail.id}>{item.menuDetail.title}</MenuItem>;
        });
        return (
            <Menu id={'menu-' + props.menuDetail.id}
                aria-labelledby={"menu-lbl-" + props.menuDetail.id}
                data-testid={'menu-' + props.menuDetail.id}
                anchorOrigin={{
                    vertical: 'top',
                    horizontal: 'left',
                }}
                transformOrigin={{
                    vertical: 'top',
                    horizontal: 'right',
                }}
                open={open}
                onClose={props.handleClose}>
                {menuItemComponent}
            </Menu>
        );
    } else {
        return null;
    }
}

export function MenuLableComponent(props: { 'id': number | string, 'title': string, 'handleClick': MouseEventHandler }) {
    return (

        <Button
            data-testid={"menu-lbl-" + props.id}
            id={"menu-lbl-" + props.id}
            aria-haspopup="true"
            aria-controls={'menu-' + props.id}
            variant="contained"
            onClick={props.handleClick}
        > {props.title}
        </Button>

    );
}

export function MenuWrapper(props: MenuItemProps) {
    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
    const open = Boolean(anchorEl);
    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
        setAnchorEl(event.currentTarget);
    };
    const handleClose = () => {
        setAnchorEl(null);
    };
    const menuDetail = {
        ...props.menuDetail,
        open: open
    };
    return (
        <Grid item >
            <MenuLableComponent id={props.menuDetail.id} key={props.menuDetail.id} handleClick={handleClick} title={props.menuDetail.title} />
            <MenuItemComponentList key={'menu-comp-key-' + props.menuDetail.id} menuDetail={menuDetail} handleClose={handleClose} />
        </Grid>
    );
}

export function MenuComponents(props: HeaderProps): React.ReactElement | null {
    let menuListComponent: Array<React.ReactElement> = [];

    if (props.menus && props.menus.length > 0) {
        menuListComponent = props.menus.map((item) => {
            return <MenuWrapper key={'menu-' + item.menuDetail.id} menuDetail={item.menuDetail} />
        });
        return (
            <Grid container spacing={{ xs: .3, md: 1 }} data-testid='menu-components'>
                {menuListComponent}
            </Grid>
        );
    } else {
        return null;
    }
}