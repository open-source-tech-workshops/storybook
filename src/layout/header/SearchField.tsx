import * as React from "react";
import InputUnstyled, { InputUnstyledProps } from "@mui/base/InputUnstyled";
import { styled } from "@mui/system";
import Autocomplete from "@mui/material/Autocomplete";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";

const SearchField = (props: any) => {
  return (
    <Box
      sx={{
        ...props.sx,
        display: "flex",
        alignItems: "center",
      }}
    >
      <Autocomplete
        id="search-autocomplete"
        sx={{
          minWidth: "200px",
          flexGrow: 1,
        }}
        options={top100Films.map((option) => option.title)}
        renderInput={(params) => (
          <TextField
            sx={{
              justifySelf: "stretch",
              backgroundColor: "white",
              borderRadius: "8px",
              height: "3rem",
            }}
            placeholder={props.placeholder}
            {...params}
          />
        )}
      />
      <Button
        variant="contained"
        sx={{
          borderRadius: "8px",
          height: "3rem",
        }}
      >
        Search
      </Button>
    </Box>
  );
};

const top100Films = [
  { title: "The Shawshank Redemption", year: 1994 },
  { title: "The Godfather", year: 1972 },
  { title: "The Godfather: Part II", year: 1974 },
  { title: "The Dark Knight", year: 2008 },
  { title: "12 Angry Men", year: 1957 },
  { title: "Schindler's List", year: 1993 },
  { title: "Pulp Fiction", year: 1994 },
  {
    title: "The Lord of the Rings: The Return of the King",
    year: 2003,
  },
  { title: "The Good, the Bad and the Ugly", year: 1966 },
  { title: "Fight Club", year: 1999 },
  {
    title: "The Lord of the Rings: The Fellowship of the Ring",
    year: 2001,
  },
  {
    title: "Star Wars: Episode V - The Empire Strikes Back",
    year: 1980,
  },
];

export default SearchField;
