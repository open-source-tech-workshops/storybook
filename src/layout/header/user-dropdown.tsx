import * as React from 'react';
import Button from '@mui/material/Button';
import Popover from '@mui/material/Popover';
import Typography from '@mui/material/Typography';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import Divider from '@mui/material/Divider';

const style = {
    width: '100%',
    maxWidth: 360,
    bgcolor: 'background.paper',
};

export const UserInfoDropDown = (props: any) => {
    return (<div>
        <Button aria-describedby={id} variant="contained" onClick={handleClick}>
            Open Popover
        </Button>
        <Popover
            id={id}
            open={open}
            anchorEl={anchorEl}
            onClose={handleClose}
            anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'left',
            }}>
            <List sx={style} component="nav" aria-label="mailbox folders">
                <ListItem button>
                    <Typography sx={{ p: 2 }}>The content of the Popover.</Typography>
                </ListItem>
                <Divider />
                <ListItem button>
                    <Button data-testid='logout-btn' variant='text'>Logout</Button>
                </ListItem>
            </List>
        </Popover></div>
    );
}