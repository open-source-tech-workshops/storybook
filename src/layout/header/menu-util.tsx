import * as React from 'react';
import { MenuItem, Menu, Button } from '@mui/material';
import { MouseEventHandler } from 'react';
import { HeaderProps } from './header';

export interface MenuItemProps {
    menuDetail: { 'id': number, 'title': string, 'open'?: boolean, 'handler': MouseEventHandler, 'menuItems'?: MenuItemProps[] }
}

interface MenuProps extends MenuItemProps {
    'handleClose': MouseEventHandler,
}

export function MenuItemComponentList(props: MenuProps): React.ReactElement | null {
    const open = Boolean(props.menuDetail.open);
    if (props.menuDetail.menuItems && props.menuDetail.menuItems.length > 0) {
        let menuItemComponent: Array<React.ReactElement> = [];
        menuItemComponent = props.menuDetail.menuItems.map((item) => {
            return <MenuItem data-testid={'menu-item-' + props.menuDetail.id + '-' + item.menuDetail.id} onClick={item.menuDetail.handler} key={item.menuDetail.id}>{item.menuDetail.title}</MenuItem>;
        });
        return (
            <Menu id={'menu-' + props.menuDetail.id}
                data-testid="basic-menu"
                open={open}
                onClose={props.handleClose}>
                {menuItemComponent}
            </Menu>
        );
    } else {
        return null;
    }
}

export function MenuLableComponent(props: { 'id': number | string, 'title': string, 'handleClick': MouseEventHandler }) {
    return (
        <Button
            data-testid={"menu-lbl-" + props.id}
            aria-haspopup="true"
            aria-controls={'menu-' + props.id}
            onClick={props.handleClick}
        > {props.title}
        </Button>
    );
}

export function MenuWrapper(props: MenuItemProps) {
    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
    const open = Boolean(anchorEl);
    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
        setAnchorEl(event.currentTarget);
    };
    const handleClose = () => {
        setAnchorEl(null);
    };
    const menuDetail = {
        ...props.menuDetail,
        open: open
    };
    return (
        <div>
            <MenuLableComponent id={props.menuDetail.id} key={props.menuDetail.id} handleClick={handleClick} title={props.menuDetail.title} />
            <MenuItemComponentList key={'menu-comp-key-'+props.menuDetail.id} menuDetail={menuDetail} handleClose={handleClose} />
        </div>
    );
}

export function MenuComponents(props: HeaderProps): React.ReactElement | null {
    let menuListComponent: Array<React.ReactElement> = [];

    if (props.menus && props.menus.length > 0) {
        menuListComponent = props.menus.map((item) => {
            return <MenuWrapper key={'menu-'+item.menuDetail.id} menuDetail={item.menuDetail} />
        });
        return (
            <div data-testid='menu-components'>
                {menuListComponent}
            </div>
        );
    } else {
        return null;
    }
}