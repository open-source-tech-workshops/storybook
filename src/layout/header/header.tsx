import React from 'react';
import './Header.css';
import AppBar from '@mui/material/AppBar';
import TextField from '@mui/material/TextField';
import Toolbar from '@mui/material/Toolbar';
import Button from '@mui/material/Button';
import IconButton from '@mui/material/IconButton';
import Box from '@mui/material/Box'
import { MenuComponents, MenuItemProps } from './MenuUtil'
import { AuthController, UserProps } from './AuthController';

export interface HeaderProps {
    searchAutoComplateList?: { label: string; id: number; }[],
    menus?: MenuItemProps[],
    userData?: UserProps,
    searchFieldHandler?: (value: string) => void
}

const Header = (props: HeaderProps) => {
    const [searchValue, setSearchValue] = React.useState('');
    const handleSearchValueChange = (event: React.ChangeEvent<any>, handler?: (value: string) => void) => {
        setSearchValue(event.target.value);
        console.log('def', handler);
        if (handler) {
            handler(event.target.value);
        }
    };

    return (<div data-testid='app-header'>
        <Box>
            <AppBar position='fixed'>
                <Toolbar>
                    <IconButton
                        size="large"
                        edge="start"
                        color="inherit"
                        aria-label="menu"
                        sx={{ mr: 2 }}> logo </IconButton>
                    <TextField
                        required 
                        label="search" variant="outlined"
                        type="search" fullWidth
                        inputProps={{ "data-testid": "header-search-fld" }}
                        value={searchValue} onChange={(e:React.ChangeEvent<Element>)=>{handleSearchValueChange(e,props.searchFieldHandler)}}
                    />
                    <Button data-testid='search-btn' variant='contained'>Search</Button>
                    <MenuComponents menus={props.menus} />
                    {props.userData != undefined ? <AuthController userData={props.userData} /> : <AuthController />}
                </Toolbar>
            </AppBar>
        </Box>
    </div>);
}

export default Header;

