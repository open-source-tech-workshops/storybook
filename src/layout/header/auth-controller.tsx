import * as React from 'react';
import { Button } from '@mui/material';
import { UserInfoDropDown } from './user-dropdown';

export interface AuthProps{
    'userData'?: UserProps
}
export interface UserProps{
    'id':string, 'name':string
}
export const AuthController = (props: AuthProps ) => {
    if(props.userData){
        return(
            <UserInfoDropDown/>
        );
    }else{
        return(
            <Button data-testid='login-btn' variant='outlined'>Login</Button>
        );
    }
}