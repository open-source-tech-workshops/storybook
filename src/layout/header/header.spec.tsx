import * as React from 'react';
import TestRenderer, { ReactTestInstance, ReactTestRenderer } from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Header } from './header';
import { ReactNode, ReactElement, MouseEventHandler } from 'react';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Autocomplete from '@mui/material/Autocomplete';
import Button from '@mui/material/Button';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';
import '@testing-library/jest-dom/extend-expect';
import { MenuLableComponent, MenuComponents, MenuItemProps } from './menu-util';



describe('Test header', () => {
    const subMenueTestFun: MouseEventHandler = () => { }
    const headerJsx: ReactElement = <Header />;

    it('should display header with default style', () => {
        const headerComponent: TestRenderer.ReactTestRendererJSON | TestRenderer.ReactTestRendererJSON[] | null = TestRenderer.create(headerJsx).toJSON();
        expect(headerComponent).toBeDefined();
        expect(headerComponent).toMatchSnapshot();
    })

    it('should display main bar', () => {
        const testComponent = TestRenderer.create(headerJsx);
        const testComponentInstance = testComponent.root;
        expect(testComponentInstance.findByType(AppBar).props.position).toBe('fixed');
        expect(testComponentInstance.findByType(Toolbar)).toBeTruthy();
    })

    it('should display search bar', () => {
        const options = [
            { label: 'The Godfather', id: 1 },
            { label: 'Pulp Fiction', id: 2 },
        ];
        const headerJsx: ReactElement = <Header searchAutoComplateList={...options} />;
        const testComponent: ReactTestRenderer = TestRenderer.create(headerJsx);
        const testComponentInstance = testComponent.root;
        expect(testComponentInstance.findByType(Autocomplete).props.options[0].label).toBe('The Godfather');
        expect(testComponentInstance.findByType(Button).props.variant).toBe('outlined');
        expect(testComponentInstance.findByType(Button).props.children).toEqual('Search');

    })

    it('should display brand log', () => {
        const testComponent: ReactTestRenderer = TestRenderer.create(headerJsx);
        const testComponentInstance = testComponent.root;
        expect(testComponentInstance.findByProps({ "size": "large", "edge": "start", "color": "inherit", "aria-label": "menu" }).props.children).toEqual(' logo ');
    })

    it('should display menu with 2 menu', () => {
        const menues: MenuItemProps[] = [
            { menuDetail: { id: 1, title: 'sub1', handler: subMenueTestFun } },
            { menuDetail: { id: 2, title: 'sub2', handler: subMenueTestFun } },
        ];
        const headerJsx: ReactElement = <Header menus={menues} />;
        const testComponent = TestRenderer.create(headerJsx);
        const testComponentInstance: ReactTestInstance = testComponent.root;
        const testMenuInstance = testComponentInstance.findByType(MenuComponents);
        const testMenuInstances: ReactTestInstance[] = testComponentInstance.findAllByType(MenuLableComponent);
        expect(testMenuInstances.length).toBe(2);
    })

    it('should display menu with 1 menu without menu-item 1 menu with 3 menu-item', () => {
        const menue2ClickFun: MouseEventHandler = jest.fn();

        const submenu: MenuItemProps[] = [
            { menuDetail: { id: 1, title: 'sub-menu 1', handler: subMenueTestFun } },
            { menuDetail: { id: 2, title: 'sub-menu 2', handler: menue2ClickFun } },
            { menuDetail: { id: 3, title: 'sub-menu 3', handler: subMenueTestFun } },

        ]
        const menues: MenuItemProps[] = [
            { menuDetail: { id: 1, title: 'menu 1', handler: subMenueTestFun } },
            { menuDetail: { id: 2, title: 'menu 2', handler: subMenueTestFun, menuItems: submenu } },
        ];
        const headerJsx: ReactElement = <Header menus={menues} />;
        const { getByTestId } = render(headerJsx);
        const menuList: HTMLElement = getByTestId('menu-components');
        expect(menuList).toBeTruthy();
        const menu2Button: HTMLElement = getByTestId('menu-lbl-2');
        expect(menu2Button).toBeTruthy();
        fireEvent.click(menu2Button);
        const menu2SubItem: HTMLElement = getByTestId('menu-item-2-2');
        expect(menu2SubItem).toBeTruthy();
        fireEvent.click(menu2SubItem);
        expect(menue2ClickFun).toHaveBeenCalledTimes(1);
    })


    // it('should display log in button for logged out user', ()=>{

    // })

    // it('should display user account drop down for logged in user', ()=>{
    //     const { getByRole } = render(headerJsx);
    //     expect(getByRole('user-account-dpd')).toBeTruthy();
    // })

    // it('should display logged out button for logged in user', ()=>{
    //     const { getByRole } = render(headerJsx);
    //     expect(getByRole('user-account-dpd')).toBeTruthy();
    // })

    // it('should display header with primary style', ()=>{

    // })

    // it('should display header with secondary style', ()=>{

    // })

})
