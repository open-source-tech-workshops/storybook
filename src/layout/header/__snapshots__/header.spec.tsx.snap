// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test header should display header with default style 1`] = `
<div
  data-testid="app-header"
>
  <div
    className="MuiBox-root css-0"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
           logo 
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            search
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputTypeSearch css-1g24dm6-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="header-search-fld"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="search"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-173wfxe"
              >
                <span>
                  search
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="search-btn"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Search
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          data-testid="login-btn"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Login
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</div>
`;
